base/applications/cleanmgr/util.c

```diff
@@ -0,0 +1,1179 @@
+/*
+ * PROJECT:         ReactOS Disk Cleanup
+ * LICENSE:         GPL-2.0+ (https://spdx.org/licenses/GPL-2.0+)
+ * PURPOSE:         Utility functions
+ * COPYRIGHT:       Copyright 2020 Arnav Bhatt (arnavbhatt288 at gmail dot com)
+ */
+ 
+#include "resource.h"
+#include "precomp.h"
+
+#define KB 1024
+#define MB 1024 * KB
+#define GB 1024 * MB
+#define TB 1024 * GB
+
+#define ALLOW_FILE_REMOVAL 2
+#define DISALLOW_FILE_REMOVAL 3
+
+#define CX_BITMAP 20
+#define CY_BITMAP 20
+
+typedef enum
+{
+    OLD_CHKDSK_FILES = 0,
+    RAPPS_FILES = 1,
+    RECYCLE_BIN = 2,
+    TEMPORARY_FILE = 3
+} DIRECTORIES;
+
+#define ETDT_DISABLE        0x00000001
+#define ETDT_ENABLE         0x00000002
+#define ETDT_USETABTEXTURE  0x00000004
+#define ETDT_ENABLETAB      (ETDT_ENABLE  | ETDT_USETABTEXTURE)
+
+typedef HRESULT (WINAPI * ETDTProc) (HWND, DWORD);
+
+DWORD WINAPI SizeCheck(LPVOID lpParam)
+{
+    HKEY hRegKey = NULL;
+    //SETTINGS_INFO SettingsInfo;
+    //ZeroMemory(&SettingsInfo, sizeof(SETTINGS_INFO));
+    //DWORD dwSize = sizeof(SettingsInfo);
+    
+    if (RegOpenKeyExW(HKEY_CURRENT_USER,
+                      L"Software\\ReactOS\\rapps",
+                      0,
+                      KEY_QUERY_VALUE,
+                      &hRegKey) != ERROR_SUCCESS)
+    {
+        DPRINT("RegOpenKeyExW(): Failed to open a registry key!\n");
+    }
+    
+    HWND hwnd = (HWND)lpParam;
+    
+    HWND hProgressBar = NULL;
+    WCHAR TargetedDir[MAX_PATH] = { 0 };
+    SHQUERYRBINFO RecycleBinInfo;
+    ZeroMemory(&RecycleBinInfo, sizeof(RecycleBinInfo));
+    RecycleBinInfo.cbSize = sizeof(RecycleBinInfo);
+
+    hProgressBar = GetDlgItem(hwnd, IDC_PROGRESS_1);
+    
+    if(bv.SysDrive == TRUE)
+    {
+        SHGetFolderPathW(NULL, CSIDL_WINDOWS, NULL, SHGFP_TYPE_CURRENT, TargetedDir);
+        StringCbCatW(TargetedDir, _countof(TargetedDir), L"\\Temp");
+        if (PathIsDirectoryW(TargetedDir))
+        {
+            sz.TempASize = DirSizeFunc(TargetedDir);
+            if (sz.TempASize == 1)
+            {
+                return FALSE;
+            }
+        }
+        SHGetFolderPathW(NULL, CSIDL_PROFILE, NULL, SHGFP_TYPE_CURRENT, TargetedDir);
+        StringCbCatW(TargetedDir, _countof(TargetedDir), L"\\Local Settings\\Temp");
```

Comment:
`GetTempPathW` or explicitly looking at the `%TEMP%` environment variable seems preferable to hardcoding this path (which, for one, differs between OS localizations and versions)

Reply:
Windows's cleanmgr reads the entries from `HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows\CurrentVersion\Explorer\VolumeCaches\Internet Cache Files` by the way (and there you can see that it just uses `%TEMP%`).

https://docs.microsoft.com/en-us/windows/win32/lwef/disk-cleanup explains how this works on Windows btw -- there's a set of COM interfaces for disk cleanup handlers.